gobject-introspection/ext/gobject-introspection/rb-gi-argument.c

```diff
@@ -2374,6 +2376,8 @@ rb_gi_return_argument_free_everything_array_c(GIArgument *argument,
     case GI_TYPE_TAG_INT32:
     case GI_TYPE_TAG_UINT32:
```

Comment:
ditto.
Or we can add support for all of `BOOLEAN`, `*INT*`, `FLOAT` and `DOUBLE`:

```c
     case GI_TYPE_TAG_BOOLEAN: 
     case GI_TYPE_TAG_INT8: 
     case GI_TYPE_TAG_UINT8:
     case GI_TYPE_TAG_INT16:
     case GI_TYPE_TAG_UINT16:
     case GI_TYPE_TAG_INT32:
     case GI_TYPE_TAG_UINT32:
     case GI_TYPE_TAG_INT64:
     case GI_TYPE_TAG_UINT64:
     case GI_TYPE_TAG_FLOAT:
     case GI_TYPE_TAG_DOUBLE:
         g_free(argument->v_pointer);
         break;

Reply:
I've added support for all of `BOOLEAN`, `*INT*`, `FLOAT` and `DOUBLE`.